soundzones_protocol/documentation_SZP.md

```diff
@@ -0,0 +1,257 @@
+| payload | 1 | - | Payload |
+
+<!--
+```
+@startuml 01_send
+server -> client: [ cid, time, payload_len, payload ]
+@enduml
+```
+-->
+
+![](sequence_diagrams/01_send.svg)
+
+
+---
+
+### A1 - enroll_c - client assigned id's
+To use this command, client has been preconfigured with a Id.
+So the rigth filters is used for the right Id.
+
+| Tag | Size [bytes] | Value | Description | 
+|---|---|---|---|
+| cid | 1 | 0xA1 | Command Id |
+|  client_id | 1 | 0x01-0xFF | Assigned client id (0 is reserved for server) |
+| res | 1 | 0 \| 1 | denie \| accept |
+
+<!--
+```
+@startuml A1_enrole_c
+server <- client: [ cid, client_id ]
+server -> client: [ cid, res ]
+@enduml
+```
+-->
+
+![](sequence_diagrams/A1_enrole_c.svg)
+
+---
+
+### A2 - enroll_s - server assigned id's
+To use this command, it has to be figured out afterwurds which client needs what filter.
+
+| Tag | Size [bytes] | Value | Description | 
+|---|---|---|---|
+| cid | 1 | 0xA2 | Command Id |
+| client_id | 1 | 0x01-0xFF | Assigned client id |
+| ack | 1 | 0x01 | Accepted |
+
+<!--
+```
+@startuml A2_enrole_s
+server <- client: [ cid ]
+server -> client: [ cid, client_id ]
+group succesful
+    server <- client: [ cid, ack]
+end
+@enduml
+```
+-->
+
+![](sequence_diagrams/A2_enrole_s.svg)
+
+---
+
+### B1 - set_time_encode
+Defines what encoding the clock uses.
+
+| Tag | Size [bytes] | Value | Description | 
+|---|---|---|---|
+| cid | 1 | 0xB1 | Command Id |
+| encoding_id | 1 | 0x01-0xFF | Encoding Id |
+| ack | 1 | 0x01 | Acknolegment |
+
+<!--
+```
+@startuml B1_set_time_encode
+server -> client: [ cid, encoding_id ]
+group succesful
+    server <- client: [ cid, ack ]
+end
+@enduml
+```
+-->
+
+![](sequence_diagrams/B1_set_time_encode.svg)
+
+**Supported** **time** **encodings**
+| Id | Name |
+|--- |--- |
+| 0x01 | hh:mm:ss:ms |
+| 0x02 | mm:ss:ms:µs |
+| 0x03 | mm:ss:ms:µs:ns |
+
+---
+
+### B2 - set_size_payload_len
+Sets the size of the payload_len.
+With a size of 2 bytes, the payload can maximum be 65.535 bytes long.
+
+| Tag | Size [bytes] | Value | Description | 
+|---|---|---|---|
+| cid | 1 | 0xB2 | Command Id |
+| size | 1 | 0x01-0xFF | Size of payload_len |
+| ack | 1 | 0x01 | Acknolegment |
+
+<!--
+```
+@startuml B2_set_time_encode
+server -> client: [ cid, size ]
+group succesful
+    server <- client: [ cid, ack ]
+end
+@enduml
+```
+-->
+
+![](sequence_diagrams/B2_set_time_encode.svg)
+
+---
+
+### B3 - set_sound_compression
+Sets the size of the payload_len.
+With a size of 2 bytes, the payload can maximum be 65.535 bytes long.
+
+| Tag | Size [bytes] | Value | Description | 
+|---|---|---|---|
+| cid | 1 | 0xB3 | Command Id |
+| compression_id | 1 | 0x01-0xFF | Payload compression type |
+| ack | 1 | 0x01 | Acknolegment |
+
+<!--
+```
+@startuml B3_set_sound_compression
+server -> client: [ cid, compression_id ]
+group succesful
+    server <- client: [ cid, ack ]
+end
+@enduml
+```
+-->
+
+![](sequence_diagrams/B3_set_sound_compression.svg)
```

Comment:
Tror vi skal lave lidt om i det her, for hvis vi sender som WAV, FLAC, PCM eller andre så er sampling frekvens og alle andre informationer om filen en del af formatet. Så i WAV sendes en header med som definere samplingrate, bitpersample, nr. of channels osv.